cmd/utils.go

```diff
@@ -332,6 +332,34 @@ func ToS3ETag(etag string) string {
 	return etag
 }
 
+func newCustomDialContext(timeout time.Duration) func(ctx context.Context, network, addr string) (net.Conn, error) {
+	return func(ctx context.Context, network, addr string) (net.Conn, error) {
+		dialer := &net.Dialer{
+			Timeout:   timeout,
+			KeepAlive: timeout,
+			DualStack: true,
+		}
+
+		return dialer.DialContext(ctx, network, addr)
+	}
+}
+
+func newCustomHTTPTransport(tlsConfig *tls.Config, timeout time.Duration) func() *http.Transport {
+	tr := &http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		DialContext:           newCustomDialContext(timeout),
+		MaxIdleConnsPerHost:   256,
+		IdleConnTimeout:       60 * time.Second,
+		TLSHandshakeTimeout:   30 * time.Second,
+		ExpectContinueTimeout: 10 * time.Second,
+		TLSClientConfig:       tlsConfig,
+		DisableCompression:    true,
```

Comment:
That's because Go http client automatically decodes gzip contentType

Reply:
Apart from that these settings don't need explanation since they are just arbitrary good values for us. It maybe good or bad we don't really know as we haven't done any specific testing for these scenarios. They are explained by Go net/http documentation on what they actually mean.